tfchain_bridge/pkg/substrate/client.go

```diff
@@ -22,17 +26,76 @@ type Versioned struct {
 
 type SubstrateClient struct {
 	*substrate.Substrate
+	Identity substrate.Identity
 }
 
 // NewSubstrate creates a substrate client
-func NewSubstrateClient(url string) (*SubstrateClient, error) {
+func NewSubstrateClient(url string, seed string) (*SubstrateClient, error) {
 	mngr := substrate.NewManager(url)
 	cl, err := mngr.Substrate()
 	if err != nil {
 		return nil, err
 	}
+	tfchainIdentity, err := substrate.NewIdentityFromSr25519Phrase(seed)
+	if err != nil {
+		return nil, err
+	}
+
+	isValidator, err := cl.IsValidator(tfchainIdentity)
+	if err != nil {
+		return nil, err
+	}
+
+	if !isValidator {
+		return nil, fmt.Errorf("account provided is not a validator for the bridge runtime")
+	}
 
 	return &SubstrateClient{
 		cl,
+		tfchainIdentity,
 	}, nil
 }
+
+func (client *SubstrateClient) SubscribeTfchainBridgeEvents(ctx context.Context, eventChannel chan EventSubscription) error {
+	defer close(eventChannel)
```

Comment:
I'd prefer for this function to not close the channel. The reason is that the channel is passed to this function, and it is technically not known to this function that it is the only sender, hence closing it here will cause a panic if there are other senders. What's more, if this function exits, and you want to open a new subscription, the caller is also required to create a new channel, and pass this new channel to the receiver.

All in all, I feel it would be better if the caller of this function (actually the owner of the channel, which is not necessarily the direct caller) closes the channel once this function returns.